src/main/java/dev/j3fftw/litexpansion/items/Wrench.java

```diff
@@ -0,0 +1,144 @@
+import io.github.thebusybiscuit.slimefun4.implementation.items.SimpleSlimefunItem;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.interfaces.InventoryBlock;
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.Arrays;
+
+/**
+ * This {@link SimpleSlimefunItem} allows players to
+ * instantly break any {@link EnergyNetComponent} block
+ * or {@link CargoNet} block.
+ *
+ * @author FluffyBear
+ *
+ */
+public class Wrench extends SimpleSlimefunItem<ItemUseHandler> implements Listener,DamageableItem {
+
+    private static final int[] cargoSlots = { 19, 20, 21, 28, 29, 30, 37, 38, 39 };
+    private static final String[] specialBlockIds = { SlimefunItems.CARGO_INPUT_NODE.getItemId(), SlimefunItems.CARGO_OUTPUT_NODE.getItemId(), SlimefunItems.CARGO_OUTPUT_NODE_2.getItemId(), SlimefunItems.CARGO_CONNECTOR_NODE.getItemId(), SlimefunItems.SMALL_CAPACITOR.getItemId(), SlimefunItems.MEDIUM_CAPACITOR.getItemId(), SlimefunItems.LARGE_CAPACITOR.getItemId(), SlimefunItems.BIG_CAPACITOR.getItemId(), SlimefunItems.CARBONADO_EDGED_CAPACITOR.getItemId(), SlimefunItems.TRASH_CAN.getItemId() };
+
+    public Wrench() {
+        super(Items.LITEXPANSION, Items.WRENCH, RecipeType.ENHANCED_CRAFTING_TABLE, new ItemStack[] {
+                Items.REFINED_IRON, null, Items.REFINED_IRON,
+                null, Items.REFINED_IRON, null,
+                null, Items.REFINED_IRON, null
+        });
+
+        Bukkit.getPluginManager().registerEvents(this, LiteXpansion.getInstance());
+    }
+
+    @Override
+    public ItemUseHandler getItemHandler() {
+        return e -> e.setUseBlock(Event.Result.DENY);
+    }
+
+    @EventHandler
+    public void onWrenchInteract(PlayerInteractEvent e) {
+        Player p = e.getPlayer();
+        // Check if player has wrench and is left clicking block
+        // Can't use offhand because a player can offhand the wrench to escape the event
+        if (isItem(e.getItem()) && !isItem(p.getInventory().getItemInOffHand())
+            && e.getAction().toString().endsWith("_BLOCK")
+        ) {
+            e.setCancelled(true);
+            Block block = e.getClickedBlock();
+            SlimefunItem slimefunBlock = BlockStorage.check(block);
+
+            // Check if slimefunBlock is not a machine or a cargo component
+            if (slimefunBlock == null
+                || (!(slimefunBlock instanceof EnergyNetComponent)
+                && !slimefunBlock.getID().startsWith("CARGO_NODE")
```

Comment:
`instanceof AbstractCargoNode`, don't rely on id formats!

Reply:
AbstractCargoNode not public? Any other way to get to it?